File: Data Structures & Algorithms/insert-new-interval/submission-1.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        res = []

        for x, y in intervals:
            if newInterval[0] > y:
                res.append([x, y])
            elif newInterval[1] < x:
                res.append(newInterval)
                newInterval = [x, y]
            else:
                newInterval[0] = min(x, newInterval[0])
                newInterval[1] = max(y, newInterval[1])
        res.append(newInterval)
        return res

File: Data Structures & Algorithms/merge-intervals/submission-1.java
class Solution {
    public int[][] merge(int[][] intervals) {
        List<int []> res = new ArrayList<>();
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        int[] curr = intervals[0];

        for (int i=1; i<intervals.length; i++) {
            int s = intervals[i][0];
            int e = intervals[i][1];

            if (s <= curr[1]) {
                curr[1] = Math.max(e, curr[1]);
            } else {
                res.add(new int[]{curr[0], curr[1]});
                curr = new int[]{s, e};
            }
        }
        res.add(curr);
        return res.toArray(new int[res.size()][]);
    }
}
